openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,273 @@
 """
 Models for the credentials service.
 """
+import os
+import uuid  # pylint: disable=unused-import
+
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.models import Site
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+
+
+MODE_CHOICES = (
+    (_('Honor'), 'honor'),
+    (_('Verified'), 'verified'),
+    (_('Professional'), 'professional'),
+)
+
+
+class AbstractCredential(TimeStampedModel):
+    """
+    Abstract Credentials configuration model.
+    """
+    site = models.ForeignKey(Site)
+    is_active = models.BooleanField(default=False)
+
+    class Meta(object):
+        abstract = True
+
+
+def assets_path(instance, filename):
+    """
+    Custom path for credentials templates and signatories file assets.
+
+    Arguments:
+        instance: CertificateTemplateAsset or Signatory object
+        filename: file to upload
+
+    Returns:
+        Path of asset file e.g.
+        credential_certificate_template_assets/1/filename,
+        signatories/1/filename
+    """
+
+    path = 'credential_certificate_template_assets'
+    if isinstance(instance, Signatory):
+        path = 'signatories'
+
+    name = os.path.join(
+        path,
+        str(instance.id),
+        filename
+    )
+    fullname = os.path.join(settings.MEDIA_ROOT, name)
```

Comment:
No! Assets should go to S3, never the server's hard drive. Our servers don't live very long.